code/modules/mob/living/carbon/human/human.dm

```diff
@@ -41,6 +41,7 @@
 	RegisterSignal(src, COMSIG_KB_RAILATTACHMENT, .proc/do_activate_rail_attachment)
 	RegisterSignal(src, COMSIG_GRAB_SELF_ATTACK, .proc/fireman_carry_grabbed) // Fireman carry
 	AddComponent(/datum/component/footstep, FOOTSTEP_MOB_HUMAN)
+	AddComponent(/datum/component/bump_attack)
```

Comment:
I don't like very much that approach. That means that every human will get the toggle action, while less than 2-5% will actually use it. We don't want to have useless actions lying around.

Even performance wise, you are checking if human can bump or not, even if they are using a minigun loadout... Doesn't make sense.

Either you manage to make the giving action and the registering only when you wield the vali harvester, or you make a bump component for the weapon and not for the living. Also, who cares about unarmed attacks, don't tackle that unless it's really easy to add and doesn't lead to shitcode

Reply:
yes that

Reply:
Made it so that the component's checks and such need to be enabled for the component to run.